src/System.Private.CoreLib/shared/System/Text/UTF32Encoding.cs

```diff
@@ -39,8 +39,8 @@ public sealed class UTF32Encoding : Encoding
         internal static readonly UTF32Encoding s_default = new UTF32Encoding(bigEndian: false, byteOrderMark: true);
         internal static readonly UTF32Encoding s_bigEndianDefault = new UTF32Encoding(bigEndian: true, byteOrderMark: true);
 
-        private static readonly byte[] s_bigEndianPreamble = new byte[4] { 0x00, 0x00, 0xFE, 0xFF };
-        private static readonly byte[] s_littleEndianPreamble = new byte[4] { 0xFF, 0xFE, 0x00, 0x00 };
+        private static ReadOnlySpan<byte> BigEndianPreamble => new byte[4] { 0x00, 0x00, 0xFE, 0xFF }; // uses C# compiler's optimization for static byte[] data
```

Comment:
Can this just be inline in the Preamble property, similar to how GetPreamble() method is done? No need to define a trivial properties for this.

Reply:
Sure.